### PR TITLE
feat(segwit): fee estimation test coverage

### PR DESCRIPTION
**Pull request by pavel-bc**
## Description

Improvements for #1779

- [x] Removed deprecated `branchAndBound` selection & related tests
- [x] Removed unsupported `MULTISIG-*` input types 
- [x] Removed weight unit multiplication in `getByteCount()` to avoid rounding issues
- [x] Increased `P2PKH` input size to 148 vbytes
- [x] Added `P2PKH` & `P2WPKH` test cases with different IO count
- [x] Added test cases with different type of inputs

## Testing Steps

```
$ npm run test:core -g "Coin*" 
```

### CODE REVIEW

**Review by mpfluger — APPROVED** [review]

**Review by matsjj — APPROVED** [review]

**Review by mpfluger — APPROVED** [review]

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 1]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+        // 10 + 148 + 34 = 192
+        expect(cs.transactionBytes([p2pkhIn], [p2pkhOut])).toEqual(192)
+      })
+      it('should return the right transaction size (1 P2PKH, 1 P2WPKH)', () => {
+        // 10 + 148 + 31 = 189
+        expect(cs.transactionBytes([p2pkhIn], [p2wpkhOut])).toEqual(189)
+      })
+      it('should return the right transaction size (1 P2WPKH, 1 P2PKH)', () => {
+        // 10.75 + 67.75 + 34 = 189
+        expect(cs.transactionBytes([p2wpkhIn], [p2pkhOut])).toEqual(112.5)
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:31]: typo

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 2]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+        )
+      })
+      it('should return the right transaction size (1 P2PKH, 1 P2PKH + 1 P2WPKH)', () => {
+        // 10 + 148 + 31 + 34 = 223
+        expect(cs.transactionBytes([p2pkhIn], [p2pkhOut, p2wpkhOut])).toEqual(
+          223
+        )
+      })
+      it('should return the right transaction size (1 P2WPKH, 2 P2PKH)', () => {
+        // 10.75 + 67.75 + 31*2 = 146.5
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:59]: `10.75 + 67.75 + 34*2`

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 3]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+      })
+    })
+
+    describe('2x1 transactions', () => {
+      it('should return the right transaction size (2 P2PKH, 1 P2PKH)', () => {
+        // 10 + 148*2 + 34 = 340
+        expect(cs.transactionBytes([p2pkhIn, p2pkhIn], [p2pkhOut])).toEqual(340)
+      })
+      it('should return the right transaction size (2 P2PKH, 1 P2WPKH)', () => {
+        // 10 + 148*2 + 31 = 220
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:84]: `10 + 148*2 + 31 = 337`

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 4]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+        )
+      })
+      it('should return the right transaction size (2 P2WPKH, 1 P2WPKH)', () => {
+        // 10.75 + 67.75*2 + 31 = 177.25
+        expect(cs.transactionBytes([p2wpkhIn, p2wpkhIn], [p2wpkhOut])).toEqual(
+          177.25
+        )
+      })
+      it('should return the right transaction size (1 P2PKH + 1 P2WPKH, 1 P2WPKH)', () => {
+        // 10.75 + 67.75 + 148 + 31 = 177.25
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:108]: `10.75 + 67.75 + 148 + 31 = 257.5`

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 5]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+
+    describe('2x2 transactions', () => {
+      it('should return the right transaction size (2 P2PKH, 2 P2PKH)', () => {
+        // 10 + 148*2 + 34*2 = 374
+        expect(
+          cs.transactionBytes([p2pkhIn, p2pkhIn], [p2pkhOut, p2pkhOut])
+        ).toEqual(374)
+      })
+      it('should return the right transaction size (2 P2PKH, 2 P2WPKH)', () => {
+        // 10 + 148*2 + 31*2 = 220
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:123]: `10 + 148*2 + 31*2 = 368`

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 6]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+        ).toEqual(294.5)
+      })
+      it('should return the right transaction size (2 P2PKH, 1 P2PKH + 1 P2WPKH)', () => {
+        // 10 + 148*2 + 31 + 34 = 371
+        expect(
+          cs.transactionBytes([p2pkhIn, p2pkhIn], [p2pkhOut, p2wpkhOut])
+        ).toEqual(371)
+      })
+      it('should return the right transaction size (1 P2PKH + 1 P2PWKH, 1 P2PKH + 1 P2WPKH)', () => {
+        // 10.75 + 67.75 + 148 + 31 + 34 = 371
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:141]: `10.75 + 67.75 + 148 + 31 + 34 = 291.5`

File: packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js [hunk 7]
```diff
@@ -5,13 +5,171 @@ import * as cs from './index'
...
+        ).toEqual(214.25)
+      })
+      it('should return the right transaction size (2 P2WPKH, 2 P2WPKH)', () => {
+        // 10.75 + 67.75*2 + 31*2 = 208.25
+        expect(
+          cs.transactionBytes([p2wpkhIn, p2wpkhIn], [p2wpkhOut, p2wpkhOut])
+        ).toEqual(208.25)
+      })
+      it('should return the right transaction size (2 P2WPKH, 1 P2PKH + 1 P2WPKH)', () => {
+        // 10.75 + 67.75*2 + 31 + 34 = 208.25
```
Comment (pavel-bc) [inline packages/blockchain-wallet-v4/src/coinSelection/coinSelection.spec.js:159]: `10.75 + 67.75*2 + 31 + 34 = 211.25`

**Pull request merged**